File: 2/2.js
function* fibonacci() {
  var a = 1, b = 1, temp;

  while (true) {
    yield a;

    temp = a;
    a = b;
    b = b + temp;
  }
}

var i = 0,
    total = 0;

for (i of fibonacci()) {
  if (i >= 4000000) break;

  if (i % 2 === 0) {
    total = total + i;
  }
}

console.log(total);

File: 2/2.py
def fib():
  a = b = 1
  while True:
    yield a
    a, b = b, a + b

fib_generator = fib()

total = 0
i = next(fib_generator)

while i < 4000000:
  if i % 2 == 0:
    total = total + i

  i = next(fib_generator)

print(total)
